### ISSUE TITLE
Service is added twice

### ISSUE THREAD

**Issue by redssu** [rohitsangwan01/ble_peripheral#4]
Hey,

I noticed that one of the services passed to the "addServices" function is being added twice. I believe there is an oversight in the "addServices" function (if it is important, I was adding a list with only one service): 

https://github.com/rohitsangwan01/ble_peripheral/blob/5f938166f31d847a88cf04fe5bebbc29c6b0f059/android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt#L97-L106

as you can see, you add all the services to the queue and then explicitly add the last one, so it looks like it is added twice. (First time with explicit add, next time with queued add)

I have never used kotlin, but I think it can be solved this way:

```kt
    override fun addServices(services: List<BleService>) {
        // To make sure that we are safe from race condition
        val totalServices = services.size - 1
        val numOfServicesToAdd = if (servicesToAdd.peek() != null) totalServices else (totalServices - 1) 
        
        for (i in 0..numOfServicesToAdd) {
            servicesToAdd.add(services[i].toGattService())
        }
        
        // Start a queue if it isn't started
        if (numOfServicesToAdd != totalServices) {
            addService(services.last().toGattService())
        }
    }
```

I tested this, and this code seems to solve the problem.

If my code is good enough, I can make PR with this change.

Thanks!

**Comment by rohitsangwan01** [comment]
@redssu seems like this is a bug, I was actually planning to leave the race condition handling on flutter side
And just provide a method to add single service, like native

**Comment by rohitsangwan01** [comment]
@redssu fixed in new version ( 1.0 ) now we have to add each service one by one, this will improve the race conditions, and proper errors handeling if any service failed to add
